File: C# Pattern Programs/Wave Patterns/Pattern 23.cs
using System;

   class Wave_23
 {
    public static void Main(string[] args)
    {

    int wave_height = 4; //change  value to increase or decrease the height of wave
    int wave_length = 4; //change  value to increase or decrease the length of wave

    int i;
    int j;
    int k;
    int num = wave_height + 1;
    int jump = (num * 3 + 2) - (num + 3);
    int inc = 1;
    int x = 0;


    for (i = 1;i <= wave_height;i++)
    {
    for (j = 1;j <= wave_length;j++)
    {
    if (i != wave_height)
    {
    Console.Write("   "); // 3 spaces
    }
    else
    {
    Console.Write("{0:D2} ", (num - 1));
    }

    for (k = 1;k <= 4;k++)
    {
    if (i == 1 || k == 1 || k == 4)
    {
    Console.Write("{0:D2} ", num);
    num = num + inc;
    }
    else
    {
    Console.Write("   "); // 3 spaces
    }


    }
    num = num - inc; //restore value of n (false increment)




    if (i != wave_height)
    {
    Console.Write("   "); // 3 spaces
    }
    else
    {
    Console.Write("{0:D2} ", (num + 1));
    }

    num += jump;

    }

    jump -= 2;
    inc = 5 + x;
    x += 2;

    num = wave_height - i + 1;

    Console.WriteLine();
    }
     Console.ReadKey(true);
    }
 }

File: PythonPatternPrograms/WavePatterns/Pattern 23.py
wH = 4  # 1
wL = 4  # 2
num = wH + 1
jump = (num * 3 + 2) - (num + 3)
inc = 1
m = 0

for x in range(1, wH + 1):
    for y in range(1, wL + 1):
        if (x != wH):
            print("   ", end="")  # 3
        else:
            print("{:02d} ".format(num - 1), end="")
        for z in range(1, 5):
            if (x == 1 or z == 1 or z == 4):
                print("{:02d} ".format(num), end="")
                num += inc
            else:
                print("   ", end="")  # 3
        num -= inc  # 4
        if (x != wH):
            print("   ", end="")  # 3
        else:
            print("{:02d} ".format(num + 1), end="")
        num += jump
    jump -= 2
    inc = 5 + m
    m += 2
    num = wH - x + 1
    print()

    """
1) wH - change value to increase/decrease the height of the wave
2) wL - change value to increase/decrease the length of the wave
3) 3 whitespaces
4) restore value of num 

    """